### PR TITLE
mPower2 migration

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
This removes the BridgeApp project from main including removing Carthage support. Additionally, it reverts the changes to the survey engine to use the deprecated classes and methods that are no longer supported in SageResearch 4.0 directly b/c they do not conform to Kotlin polymorphic serialization. All unit tests have been migrated to run in SwiftPM with an iOS target and a GitHub action was added to run those tests when issuing a pull request against main.

I have not finished migrating mPower2 and this should not be tagged. Once it is tagged, that tagging should bump the major version. While mPower2 is the only app that uses this feature AT THIS TIME, I am not sure we won't get requests to integrate MTB into other applications and this allows supporting data tracking and Bridge 1.0 surveys. Also, even if we were to decide *not* to ever pull this into main, I didn't want to leave this indefinitely w/o any review since the changes are fairly substantial.

### CODE REVIEW

File: DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationEditDetailsViewController.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ import UIKit
 import Research
 import ResearchUI
 
-protocol SBAMedicationEditDetailsViewControllerDelegate : class {
+protocol SBAMedicationEditDetailsViewControllerDelegate : AnyObject {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationEditDetailsViewController.swift:39]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationFollowupTask.swift [hunk 2]
```diff
@@ -33,6 +33,8 @@
 
 import Foundation
 import Research
+import MobilePassiveData
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationFollowupTask.swift:36]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationLoggingStepObject.swift [hunk 3]
```diff
@@ -86,8 +86,8 @@ open class SBAMedicationLoggingStepObject : SBATrackedSelectionStepObject {
         _commonInit()
     }
     
-    override public init(identifier: String, items: [SBATrackedItem], sections: [SBATrackedSection]? = nil, type: RSDStepType? = nil) {
-        super.init(identifier: identifier, items: items, sections: sections, type: type)
+    override public init(identifier: String, items: [SBATrackedItem], sections: [SBATrackedSection]? = nil) {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationLoggingStepObject.swift:89]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationReminderManager.swift [hunk 4]
```diff
@@ -210,7 +210,7 @@ open class SBAMedicationReminderManager : SBAScheduleManager, UNUserNotification
                             self.addNotifications(for: medicationResult)
                         }
                     }
-                case .authorized, .provisional:
+                case .authorized, .provisional, .ephemeral:
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationReminderManager.swift:213]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationTrackingStepNavigator.swift [hunk 5]
```diff
@@ -39,6 +39,10 @@ import BridgeSDK
 
 open class SBAMedicationTrackingStepNavigator : SBATrackedItemsStepNavigator {
     
+    open override class func defaultType() -> RSDTaskType {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBAMedicationTrackingStepNavigator.swift:42]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBATrackedMedicationLoggingStepViewController.swift [hunk 6]
```diff
@@ -162,7 +162,7 @@ extension SBATrackedMedicationLoggingStepViewController: SBAMedicationLoggingCel
     }
 }
 
-public protocol SBAMedicationLoggingCellDelegate : class, NSObjectProtocol {
+public protocol SBAMedicationLoggingCellDelegate : AnyObject {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBATrackedMedicationLoggingStepViewController.swift:165]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBATrackedMedicationReviewStepObject.swift [hunk 7]
```diff
@@ -34,10 +34,17 @@
 import Foundation
 import UIKit
 import Research
+import ResearchUI
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBATrackedMedicationReviewStepObject.swift:37]: Copyright date?

File: DataTracking/DataTracking/iOS/Medication Tracking/SBAWarningViewController.swift [hunk 8]
```diff
@@ -36,7 +36,7 @@ import Research
 import ResearchUI
 import UIKit
 
-protocol SBAWarningViewControllerDelegate : class {
+protocol SBAWarningViewControllerDelegate : AnyObject {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Medication Tracking/SBAWarningViewController.swift:39]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedItem.swift [hunk 9]
```diff
@@ -35,6 +35,7 @@ import Foundation
 import Research
 import BridgeApp
 import BridgeSDK
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedItem.swift:38]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedItemRemindersStepObject.swift [hunk 10]
```diff
@@ -34,29 +34,34 @@
 import Foundation
 import Research
 import ResearchUI
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedItemRemindersStepObject.swift:37]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedItemsLoggingStepObject.swift [hunk 11]
```diff
@@ -41,6 +41,10 @@ import BridgeSDK
 /// information about a list of tracked items for each one.
 open class SBATrackedItemsLoggingStepObject : SBATrackedSelectionStepObject {
     
+    open override class func defaultType() -> RSDStepType {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedItemsLoggingStepObject.swift:44]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedItemsStepNavigator.swift [hunk 12]
```diff
@@ -33,13 +33,24 @@
 
 import Foundation
 import Research
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedItemsStepNavigator.swift:36]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedLoggingDataSource.swift [hunk 13]
```diff
@@ -33,7 +33,8 @@
 
 import Foundation
 import Research
-
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedLoggingDataSource.swift:36]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedSelectionDataSource.swift [hunk 14]
```diff
@@ -33,7 +33,7 @@
 
 import Foundation
 import Research
-
+import BridgeApp
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedSelectionDataSource.swift:36]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackedSelectionStepObject.swift [hunk 15]
```diff
@@ -35,6 +35,7 @@ import Foundation
 import Research
 import BridgeApp
 import BridgeSDK
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackedSelectionStepObject.swift:38]: Copyright date?

File: DataTracking/DataTracking/iOS/Model/SBATrackingDataSource.swift [hunk 16]
```diff
@@ -33,6 +33,7 @@
 
 import Foundation
 import Research
+import BridgeApp
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Model/SBATrackingDataSource.swift:36]: Copyright date?

File: DataTracking/DataTracking/iOS/Resources/en.lproj/Localizable.strings [hunk 17]
```diff
@@ -0,0 +1,114 @@
+/* 
+  Localizable.strings
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Resources/en.lproj/Localizable.strings:2]:
```suggestion
 Localizable.strings
```

File: DataTracking/DataTracking/iOS/SBADataTrackingFactory.swift [hunk 18]
```diff
@@ -33,15 +33,17 @@
 
 import Foundation
 import Research
+import ResearchUI
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/SBADataTrackingFactory.swift:36]: Copyright date?

File: DataTracking/DataTracking/iOS/Symptom Tracking/SBASymptomLoggingStepObject.swift [hunk 19]
```diff
@@ -36,10 +36,15 @@ import UIKit
 import JsonModel
 import Research
 import ResearchUI
+import BridgeApp
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Symptom Tracking/SBASymptomLoggingStepObject.swift:39]: Copyright date?

File: DataTracking/DataTracking/iOS/Symptom Tracking/SBASymptomLoggingStepViewController.swift [hunk 20]
```diff
@@ -134,7 +134,7 @@ extension SBASymptomLoggingStepViewController : SBASymptomLoggingCellDelegate {
     }
 }
 
-public protocol SBASymptomLoggingCellDelegate : class, NSObjectProtocol {
+public protocol SBASymptomLoggingCellDelegate : AnyObject {
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Symptom Tracking/SBASymptomLoggingStepViewController.swift:137]: Copyright date?

File: DataTracking/DataTracking/iOS/Triggers Tracking/SBATriggerResult.swift [hunk 21]
```diff
@@ -34,12 +34,13 @@
 import Foundation
 import JsonModel
 import Research
+import BridgeApp
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/Triggers Tracking/SBATriggerResult.swift:37]: Copyright date?

File: DataTracking/DataTracking/iOS/View Controllers/SBATrackedItemRemindersStepViewController.swift [hunk 22]
```diff
@@ -35,6 +35,8 @@ import UIKit
 import UserNotifications
 import Research
 import ResearchUI
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTracking/iOS/View Controllers/SBATrackedItemRemindersStepViewController.swift:38]: Copyright date?

File: DataTracking/DataTrackingTests/ArchivableTrackingTests.swift [hunk 23]
```diff
@@ -36,6 +36,7 @@ import XCTest
 @testable import DataTracking
 import JsonModel
 import Research
+import BridgeSDK
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/ArchivableTrackingTests.swift:39]: Copyright date?

File: DataTracking/DataTrackingTests/CodableTrackedDataTests.swift [hunk 24]
```diff
@@ -35,6 +35,8 @@ import XCTest
 @testable import BridgeApp
 @testable import DataTracking
 import Research
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/CodableTrackedDataTests.swift:38]: Copyright date?

File: DataTracking/DataTrackingTests/MedicationLoggingDataSourceTests.swift [hunk 25]
```diff
@@ -129,7 +129,7 @@ class MedicationLoggingDataSourceTests: XCTestCase {
         let timeOfDay = buildDate(weekday: .monday, hour: 10, minute: 5)
         let result = buildMedicationResult(identifier: "review")
         let meds = buildMedicationItems()
-        let step = SBAMedicationLoggingStepObject(identifier: "logging", items: meds.items, sections: meds.sections, type: .logging)
+        let step = SBAMedicationLoggingStepObject(identifier: "logging", items: meds.items, sections: meds.sections)
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/MedicationLoggingDataSourceTests.swift:132]: Copyright date?

File: DataTracking/DataTrackingTests/MedicationTrackingNavigationTests.swift [hunk 26]
```diff
@@ -35,6 +35,7 @@ import XCTest
 @testable import BridgeApp
 @testable import DataTracking
 import Research
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/MedicationTrackingNavigationTests.swift:38]: Copyright date?

File: DataTracking/DataTrackingTests/TrackedLoggingDataSourceTests.swift [hunk 27]
```diff
@@ -35,6 +35,7 @@ import XCTest
 @testable import BridgeApp
 @testable import DataTracking
 import Research
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/TrackedLoggingDataSourceTests.swift:38]: Copyright date?

File: DataTracking/DataTrackingTests/TrackedSelectionDataSourceTests.swift [hunk 28]
```diff
@@ -142,9 +142,8 @@ class TrackedSelectionDataSourceTests: XCTestCase {
     
     func buildDataSource(usingSections: Bool = true) -> RSDTableDataSource? {
         let (items, sections) = buildMedicationItems()
-        let medTracker = SBAMedicationTrackingStepNavigator(identifier: "Test", items: items, sections: usingSections ? sections : nil)
-        let task = RSDTaskObject(identifier: "medication", stepNavigator: medTracker)
-        let taskPath = RSDTaskViewModel(task: task)
+        let medTracker = SBAMedicationTrackingStepNavigator(identifier: "medication", items: items, sections: usingSections ? sections : nil)
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/TrackedSelectionDataSourceTests.swift:145]: Copyright date?

File: DataTracking/DataTrackingTests/TrackingNavigatorTests.swift [hunk 29]
```diff
@@ -33,6 +33,7 @@
 
 import XCTest
 import Research
+import JsonModel
```
Comment (Erin-Mounts) [inline DataTracking/DataTrackingTests/TrackingNavigatorTests.swift:36]: Copyright date?

**Review by Erin-Mounts — APPROVED** [review]
A bunch of nits about files edited possibly without updating the copyright year, and one missing license comment block, otherwise looks good.

File: BridgeApp/BridgeApp_UnitTest/BridgeTestHarness.swift [hunk 30]
```diff
@@ -0,0 +1,58 @@
+//
+//  BridgeTestHarness.swift
+//  
+//
+//  Created by Shannon Young on 11/11/21.
```
[the remark this answers is not in the document]
Comment (syoung-smallwisdom) [reply BridgeApp/BridgeApp_UnitTest/BridgeTestHarness.swift]: oops, missed that

**Pull request merged**